src/cdk/overlay/position/cover-position-strategy.ts

```diff
@@ -0,0 +1,608 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {coerceCssPixelValue} from '@angular/cdk/coercion';
+import {Platform} from '@angular/cdk/platform';
+import {ViewportRuler} from '@angular/cdk/scrolling';
+import {Subscription} from 'rxjs';
+
+import {OverlayReference} from '../overlay-reference';
+
+import {
+  BoundingBox,
+  boundingBoxClass,
+  BoundingBoxRect,
+  clearBoundingBoxStyles,
+  extendStyles,
+  FlexibleConnectedPositionStrategyOrigin,
+  getOriginRect,
+  resetBoundingBoxStyles,
+  resetOverlayElementStyles
+} from './flexible-positioning';
+import {PositionStrategy} from './position-strategy';
+
+/**
+ * A strategy for positioning overlays. Using this strategy, an overlay is given an
+ * implicit position relative some origin element. The relative position is defined in terms of
+ * a point on the origin element that is connected to a point on the overlay element. For example,
+ * a basic dropdown is connecting the bottom-left corner of the origin to the top-left corner
+ * of the overlay.
+ */
+export class CoverPositionStrategy implements PositionStrategy {
+  /** The overlay to which this strategy is attached. */
+  private _overlayRef: OverlayReference;
+
+  /** Whether we're performing the very first positioning of the overlay. */
+  private _isInitialRender: boolean;
+
+  /** Whether the overlay can be pushed on-screen on the initial open. */
+  private _canPush = true;
+
+  /** Whether the overlay can grow via flexible width/height after the initial open. */
+  private _growAfterOpen = false;
```

Comment:
Does growing make sense for the cover strategy? We added this feature to the connected strategy for cases like menu where you may want the overlay to grow based on the content, but I don't really know what that would be for cover.

Reply:
Note that when you don't anchor all four sides, what flows out can grow freely based on the content.

For instance, in the table edit I'll be anchoring on the top, left, and right sides but leaving the bottom unbounded so that the popup grows as tall as needed to fit the lens content.